l5kit/l5kit/sampling/agent_sampling.py

```diff
@@ -79,12 +80,16 @@ def generate_agent_sample(
         target for that state. If you sample near the end of a scene, this may contain zeroes.
 
     """
+    agent_index_start = frames[0]["agent_index_interval"][0]
+
     #  the history slice is ordered starting from the latest frame and goes backward in time., ex. slice(100, 91, -2)
     history_slice = get_history_slice(state_index, history_num_frames, history_step_size, include_current_state=True)
     history_frames = frames[history_slice]
+    history_agents = [agents[slice(*(hf["agent_index_interval"] - agent_index_start))] for hf in history_frames]
```

Comment:
Can we extract this into some function? It is a lot of copy paste. The worst part is that it is copy pasted into the test as well, so if you made a mistake in here, it would not get caught by the test. 

Ideally, write a test for this function as well to make sure you get all the frames as you expect.

Reply:
on it :)